www/scripts/codecheckerviewer/filter/BugFilterView.js

```diff
@@ -70,61 +125,79 @@ function (declare, lang, Deferred, dom, topic, Button, ContentPane, hashHelper,
       this.register(this._reportCount);
       this.addChild(this._reportCount);
 
-      //--- Run filter ---//
+      //--- Baseline filter wrapper ---//
 
-      if (this.diffView) {
-        this.cmpData = new CC_OBJECTS.CompareData();
+      var baselineFilterToggle = new FilterToggle({ title : 'Baseline' });
+      this.addChild(baselineFilterToggle);
 
-        //--- Run baseline filter ---//
+      //--- Run baseline filter ---//
 
-        this._runBaseLineFilter = new RunBaseFilter({
-          class : 'baseline',
-          title : 'Baseline',
-          parent : this,
-          updateReportFilter : function () {
-            that.runIds = this.getRunIds();
-          }
-        });
-        this.register(this._runBaseLineFilter);
-        this.addChild(this._runBaseLineFilter);
+      this._runBaseLineFilter = new RunBaseFilter({
+        class : 'run',
+        title : 'Run name',
+        parent : this,
+        updateReportFilter : function () {
+          that.runIds = this.getRunIds();
+        }
+      });
+      this.register(this._runBaseLineFilter);
+      baselineFilterToggle.addChild(this._runBaseLineFilter);
+
+      //--- Run history tags filter ---//
+
+      this._runHistoryTagFilter = new RunHistoryTagFilter({
+        class : 'run-tag',
+        title : 'Run tag',
+        parent   : this,
+        updateReportFilter : function () {
+          that.reportFilter.runTag = this.getTagIds();
+        }
+      });
+      this.register(this._runHistoryTagFilter);
+      baselineFilterToggle.addChild(this._runHistoryTagFilter);
+
+      this._newCheckFilterToggle = new FilterToggle({
+        title : 'Newcheck',
```

Comment:
What do you think if we would name it `Diff to` or `Compare to`?

Reply:
- I changed the popup message.
- As we discussed personally, I do not rename 'Newcheck' to 'Compare to'.
- It is better to handle filter items independently. With #1346 it will be possible to diff results by multiple run tags. Let's assume the user selects multiple run and tag filters for baseline and multiple run and tag filter for newcheck. If the user clears the run filter set for newcheck we do not want to change the diff type filter because multiple run tag filters are selected to newcheck.